lib/connection/result/column.js

```diff
@@ -552,22 +552,16 @@ function convertRawVariant(rawColumnValue, column, context)
   {
     try
     {
-      ret = betterEval("(" + rawColumnValue + ")");
+      ret = GlobalConfig.jsonColumnVariantParser(rawColumnValue);
     }
     catch (parseError)
     {
-      // check if raw string is in XML format
-      // ensure each tag is enclosed and all attributes and elements are valid
-      if (XMLValidator.validate(rawColumnValue) === true)
-      {
-        // use XML parser
-        ret = new XMLParser().parse(rawColumnValue);
-      }
-      else
-      {
-        // TODO: log the error
+      ret = GlobalConfig.xmlColumnVariantParser(rawColumnValue);
```

Comment:
Oops sorry I didn't realize I deleted it. I saw that comment but I wasn't sure whether to keep the logging/catch inside the default xml parser function since the default parser could be replaced

I thought about wrapping it in its own try/catch so it doesn't rely on the custom function to do the logging/catching. Do you think a nested try/catch for the xml parsing is ok or keeping it in the default function is enough?

Reply:
it's hard question... but maybe we should keep it consistent with json parser and log also here

Maybe we should have such logic here (is pseudo js):

```
try {
  return parseJson()
} catch (jsonParseError) {
  try {
     return parseXml()
  } catch(xmlParseError) {
    log(variant cannot be parsed neither as JSON (jsonParseError.message) nor as XML (xmlParseError.message)
    throw new VariantParseError(jsonParseError, xmlParseError)
  }
}
```

@sfc-gh-igarish WDYT?

Reply:
This could work, info on both error messages is provided through the logging and VariantParseError